src/main/java/mate/academy/spring/boot/intro/repository/BookRepositoryImpl.java

```diff
@@ -0,0 +1,51 @@
+package mate.academy.spring.boot.intro.repository;
+
+import java.util.List;
+import mate.academy.spring.boot.intro.model.Book;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
+
+    @Override
+    public Book save(Book book) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(book);
+            transaction.commit();
+            return book;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't insert book into DB: " + book);
```

Comment:
for DAO layer use DataProcessingException
```suggestion
            throw new DataProcessingException("Can't insert book into DB: " + book);
```